Dodge.js

```diff
@@ -1,178 +1,201 @@
 import { StatusBar } from 'expo-status-bar';
+import React, { Component, PureComponent } from 'react';
 import { StyleSheet, Text, View, Dimensions, Button, Image } from 'react-native';
-import { Knight, Circle, Brick, Grenade, MoveKnight, Shrapnel, counter} from "./Entities.js";
+import {
+  Knight, Circle, Brick, Grenade, MoveKnight,
+  Shrapnel, Water
+} from "./Entities.js";
 import { GameEngine } from "react-native-game-engine";
+import Matter from "matter-js";
 import AxisPad from 'react-native-axis-pad';
 ////this looks interesting for the attacker placing/throwing projectiles:
 //https://reactnative.dev/docs/panresponder 
 //https://medium.com/yapsody-engineering/hooks-and-function-components-in-react-native-d4e667c90cda
-var ents = {}
+let score = 0
+export default function DodgeBallGame({ navigation }) {
```

Comment:
Having learned slightly more, this is definitely unnecessary.

It should work to have DodgeBall passed directly to the Navigator:
```
<Stack.Screen 
          name="Dodge"
          component={DodgeBall}
          options={{headerShown: false}}
        />
```
Note that component is given a reference to the Class (i.e. not a specific instance of the minigame, but the general concept, from which Stack.Screen presumably makes a specific instance). When Stack.Screen creates the component, it passes in the navigator as a prop (in the current code, this is what the DodgeBallGame function sees).

So if you pass the DodgeBall directly to the Navigator, you can just call
```
class DodgeBall {
  ...
  function EndGame() {  // or wherever else you might want to navigate from
    this.props.navigation.navigate("Battle");
  }
}
```